### PR TITLE
configure user directory

### PR DESCRIPTION
**Pull request by kendonB**
# Allow the user to configure their user directory via an environment variable

## Description

This pull request allows the user to configure their user directory via the environment variable `CASTER_USER_DIR`. 

## Related Issue

Issue #842.

## Motivation and Context

See #842

## How Has This Been Tested

On Ubuntu, I ran `CASTER_USER_DIR=~/.local/share/caster-dev /usr/bin/python3 -m dragonfly load _*.py --engine kaldi  --no-recobs-messages`

and manually checked that the user directory was created correctly.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the CONTRIBUTING document.
- [x] My code follows the code style of this project.
- [x] I have checked that my code does not duplicate functionality elsewhere in Caster.
- [x] My code implements all the features I wish to merge in this pull request.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests pass.

## Maintainer/Reviewer Checklist

- [ ] Basic functionality has been tested and works as claimed.
- [ ] New documentation is clear and complete.
- [ ] Code is clear and readable.

### CODE REVIEW

File: castervoice/lib/settings.py [hunk 1]
```diff
@@ -481,7 +481,10 @@ def initialize():
     # calculate prerequisites
     SYSTEM_INFORMATION = _get_platform_information()
     _BASE_PATH = str(Path(__file__).resolve().parent.parent)
-    _USER_DIR = user_data_dir(appname="caster", appauthor=False)
+    if os.getenv("CASTER_USER_DIR") is not None:
+        _USER_DIR = os.getenv("CASTER_USER_DIR")
```
Comment (LexiconCode) [inline castervoice/lib/settings.py:485]: Does `CASTER_USER_DIR` create a permanent entry or is it only temporary for while casters running?

Comment (kendonB) [reply castervoice/lib/settings.py:485]: os.getenv does not create an entry. It just looks for and reads an existing environment variable named `CASTER_USER_DIR`.

That environment variable can exist temporarily or be set permanently.

**Comment by kendonB** [comment]
@LexiconCode, if you are happy with this, I will go ahead and make some documentation.

**Comment by LexiconCode** [comment]
> 
> 
> @LexiconCode, if you are happy with this, I will go ahead and make some documentation.

All right

**Comment by kendonB** [comment]
@LexiconCode have a read of the new documentation. I have made some edits to the existing content there.

**Comment by kendonB** [comment]
@LexiconCode is this one good to go?

**Pull request merged**